R/node.R

```diff
@@ -8,12 +8,37 @@ system_yarn <- function(..., status_ok = 0) {
   }
 }
 
-add_node <- function() {
+link_root_mklink <- function() {
+  tryCatch({
+    system2("cmd.exe", input = "mklink root ..\\..", stdout = TRUE, stderr = TRUE)
+  }, error = function(error) {
+    cli::cli_abort(c(
+      "Node.js setup: Failed to create root link with {.code mklink}: {error$message}",
+      i = "You might need to enable Developer Mode: https://go.appsilon.com/rhino-windows"
+    ))
+  })
+}
+
+link_root_fs <- function() {
+  tryCatch({
+    fs::link_create(path = fs::path("..", ".."), new_path = "root")
+  }, error = function(error) {
+    cli::cli_abort(
+      "Node.js setup: Failed to create root link with {.pkg fs}: {error$message}"
+    )
+  })
+}
+
+add_node <- function(clean = FALSE) {
+  if (clean && fs::dir_exists(node_path())) {
+    fs::dir_delete(node_path())
+  }
+
   copy_template("node", node_path())
-  fs::link_create(
-    path = fs::path("..", ".."),
-    new_path = fs::path(node_path(), "root")
-  )
+  withr::with_dir(node_path(), {
+    if (Sys.info()[["sysname"]] == "Windows") link_root_mklink()
```

Comment:
```
    Sys.info()[["sysname"]] == "Windows")
```
probably it'd make sense to extract this to a function like `detect_os` or something

Reply:
Done.